toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -0,0 +1,87 @@
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// Extract all partitions of connected image into separate files with specified format.
+func extractPartitions(imageConnection *ImageConnection, outputImageFile string, partitionFormat string) error {
+
+	// Extract basename from outputImageFile. E.g. if outputImageFile is "image.qcow2", then basename is "image".
+	basename := strings.TrimSuffix(filepath.Base(outputImageFile), filepath.Ext(outputImageFile))
+
+	// Get path of loop device associated with the image.
+	imageLoopDevice := imageConnection.Loopback().DevicePath()
+
+	// Get output directory path.
+	outDir := filepath.Dir(outputImageFile)
+
+	var err error
+
+	// Get partition info.
+	diskPartitions, err := diskutils.GetDiskPartitions(imageLoopDevice)
+	if err != nil {
+		return err
+	}
+
+	// Start extracting from 1 because the diskPartitions[0] refers to the image itself.
```

Comment:
In the `GetDiskPartitions()` implementation, if you add `TYPE` to the `--output` of the `lsblk` call, then you'll be able to distinguish between disks and partitions.

Reply:
Implemented, please take a look